src/view.c

```diff
@@ -68,19 +70,30 @@ struct view_query *
 view_query_create(void)
 {
 	struct view_query *query = znew(*query);
-	query->window_type = -1;
+	*query = (struct view_query) {
```

Comment:
Can we do this against a heap allocated struct? Feels like `*query` will point to this static memory, which doesn't seem right if we have multiple queries and actually later change one of the members.

@ahesford @jlindgren90 - Does this construct make sense (and I'm just having a wobbly) or should it be changed?

Reply:
The object on the right is stack allocated and copied into the memory pointed to by `query` (much like calling `memcpy`). It's also very likely the compiler will optimize it completely and write to the heap object directly.

Reply:
Got it. Need to get more used to writing things that way :smile: